### ISSUE TITLE
Change Reques: make custom rule `meta.messages` a required property

### ISSUE THREAD

**Issue by ren-yamanashi** [eslint/eslint#19861]
### ESLint version

v9.28.0

### What problem do you want to solve?

In the custom rule `context.report`, the `messageId` property is mandatory, but the `meta.message` property is not.

Here, specifying the `messageId` property without specifying `meta.message` (as in the following example) will result in the following error.

```ts
import { Rule } from 'eslint';

export const noClass: Rule.RuleModule = {
  meta: {
    type: 'suggestion',
  },
  create(context) {
    return {
      ClassDeclaration(node) {
        context.report({
          node,
          messageId: 'sampleId',
        });
      },
    };
  },
};
```

**_Error: Caught an error while linting    
Caused by: TypeError: context.report() called with a messageId, but no messages were present in the rule metadata._**

### What do you think is the correct solution?

Make the `meta.message` property a required property

### Participation

- [x] I am willing to submit a pull request for this change.

**Comment by mdjermanovic** [comment]
Hi @ren-yamanashi, thanks for the issue!

> In the custom rule `context.report`, the `messageId` property is mandatory, but the `meta.message` property is not.

Either `messageId` or `message` should be specified, so the `messageId` property is not mandatory. For example, this works:

```ts
import { Rule } from 'eslint';

export const noClass: Rule.RuleModule = {
  meta: {
    type: 'suggestion',
  },
  create(context) {
    return {
      ClassDeclaration(node) {
        context.report({
          node,
          message: 'My message.',
        });
      },
    };
  },
};
```

Missing `meta.messages` (or missing particular `messageId` key in `meta.messages`) will be caught by [`RuleTester`](https://eslint.org/docs/latest/integrate/nodejs-api#ruletester) when a test case produces a `context.report()` call with `messageId`.
